types/mochawesome/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for mochawesome 6.2
+// Project: https://github.com/adamgruber/mochawesome#readme
+// Definitions by: Chris Gilardi <https://github.com/Christop406>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface ReporterOptions {
+    quiet: boolean;
+    code: boolean;
+    'no-code': boolean;
+    html: boolean;
+    json: boolean;
+    consoleReporter: string;
+}
+
+interface ConfigOptions {
+    inlineDiffs?: boolean;
+    reporterOptions?: Partial<ReporterOptions>;
+}
+
+/**
+ * Initialize a new reporter.
+ *
+ * @api public
+ */
+declare function Mochawesome(runner: Mocha.Runner, options?: ConfigOptions): void;
```

Comment:
You should wrap it all in a `namespace` for the type declarations to become importable:
```suggestion
declare namespace Mochawesome {
    interface ReporterOptions {
        quiet: boolean;
        code: boolean;
        'no-code': boolean;
        html: boolean;
        json: boolean;
        consoleReporter: string;
    }

    interface ConfigOptions {
        inlineDiffs?: boolean;
        reporterOptions?: Partial<ReporterOptions>;
    }
}


/**
 * Initialize a new reporter.
 *
 * @api public
 */
declare function Mochawesome(runner: Mocha.Runner, options?: Mochawesome.ConfigOptions): void;
```